src/test/java/org/apache/commons/cli/bug/BugCLI312Test.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.junit.Test;
+
+/**
+ * Demonstrates inconsistencies in parsing Java property-style options.
+ */
+public class BugCLI312Test {
+    @Test
+    public void testPropertyStyleOption_withGetOptionProperties() throws ParseException {
+        final Option o1 = Option.builder().option("D").longOpt("define").numberOfArgs(2).valueSeparator('=').build();
+
+        final Options options = new Options();
+        options.addOption(o1);
+
+        final CommandLineParser parser = new DefaultParser();
+
+        final CommandLine cl = parser.parse(options, "-Dv -Dw=1 -D x=2 -D y -D z=3 other".split(" "));
+        assertArrayEquals(new String[] {"v", "w", "1", "x", "2", "y", "z", "3"}, cl.getOptionValues('D'));
+
+        Properties properties = cl.getOptionProperties("D");
+        assertEquals("true", properties.getProperty("v"));
+        assertEquals("1", properties.getProperty("w"));
+        assertEquals("2", properties.getProperty("x"));
+        assertEquals("true", properties.getProperty("y"));
+        assertEquals("3", properties.getProperty("z"));
+        assertEquals(5, properties.size());
+        assertEquals("other", cl.getArgList().get(0));
+    }
+
+    @Test
+    public void testPropertyStyleOption_withGetOptions() throws ParseException {
+        final Option o1 = Option.builder().option("D").longOpt("define").numberOfArgs(2).valueSeparator('=').build();
+
+        final Options options = new Options();
+        options.addOption(o1);
+
+        final CommandLineParser parser = new DefaultParser();
+
+        final CommandLine cl = parser.parse(options, "-Dv -Dw=1 -D x=2 -D y -D z=3 other".split(" "));
+        assertArrayEquals(new String[] {"v", "w", "1", "x", "2", "y", "z", "3"}, cl.getOptionValues('D'));
+
+        int defineOptionsFound = 0;
+        for (final Option o : cl.getOptions()) {
+            if ("D".equals(o.getOpt())) {
+                defineOptionsFound++;
+
+                if (defineOptionsFound == 1) {
+                    assertArrayEquals(new String[] {"v"}, o.getValues());
+                } else if (defineOptionsFound == 2) {
+                    assertArrayEquals(new String[] {"w", "1"}, o.getValues());
+                } else if (defineOptionsFound == 3) {
+                    assertArrayEquals(new String[] {"x", "2"}, o.getValues());
+                } else if (defineOptionsFound == 4) {
+                    assertArrayEquals(new String[] {"y"}, o.getValues());
+                } else if (defineOptionsFound == 5) {
+                    assertArrayEquals(new String[] {"z", "3"}, o.getValues());
+                } else {
+                    fail("Didn't expect " + defineOptionsFound + " occurrences of -D");
+                }
+            }
+        }
+        assertEquals("other", cl.getArgList().get(0));
+    }
+
+    @Test
+    public void testNoOptionValues() throws ParseException {
```

Comment:
`ParseException` is not thrown by this method, please remove it from the method signature.